sw/airborne/modules/px4_flash/px4_flash.c

```diff
@@ -130,8 +134,11 @@ void px4flash_event(void)
     unsigned char target = FLASH_PORT->get_byte(FLASH_PORT->periph);
     if (target == '1') { //target ap
       //the target is the ap, so reboot to PX4 bootloader
+#if defined(CHIBIOS_MCU_ARCH_H)
+      NVIC_SystemReset();
```

Comment:
There is now an easier way to do this:
```mcu_reboot(MCU_REBOOT_BOOTLOADER)```

For the stm32 arch this needs to be added still, but is relatively easy to add to `mcu_arch`.

Reply:
Ok, I'll fix that in a new commit. I'll also look into stm32 arch and try to rewrite that, the way you mentioned.